### DISCUSSION TITLE
Receiving HTTP 415 "Media Type Unsupported" with Got but not with Request

### DISCUSSION THREAD

**Discussion by aldipower** [sindresorhus/got#1617] [Q&A] [answered]
Hi folks.

I am doing successfully get requests on a foreign API with the following plain Request-flow (I've removed error handling for this example):

```JavaScript
function request(options) {
  return new Promise((resolve, reject) => {
    const req = https.request(options.url, {
        ...options,
      }, (res) => {

      res.on('data', d => {
        resolve(d);
      });

      req.end();
    });
});
```

I am including an oAuth-Authorization-header which looks like this:

```JavaScript
const authHeader = { authorization: 'OAuth oauth_consumer_key="xxxxxx-xxxx-xxxxx-xxxx-xxxxxxxx", oauth_nonce="xxxxxxxxxx", oauth_signature="xxxxxxxxxxxxxxxx%3D", oauth_signature_method="HMAC-SHA1", oauth_timestamp="1612448019", oauth_token="xxxxxxxxxxxxxxxxxxxxxxxxxxxxxx", oauth_version="1.0"' };
```
Calling the endpoint:
```JavaScript
const resultString = await request({
  url: "https://api.mygarmin.com/",
  method: "GET",
  headers: authHeader,
});
```

I expext a JSON string, so I do:
```JavaScript
const jsonResult = JSON.parse(resultString);
```

This works as expect. But of course is not ideal. I would like to use Got and using the retry mechanism and so on.

When I try the same with Got I am receiving and HTTP 415 Media Type Unssuproted from the API.

I've tried those settings:

```JavaScript
  const response = await got(url, {
    timeout: 10000,
    retry: 2,
    headers: {
      ...authHeader,
      accept: "application/json",
    },
    responseType: 'json',
  });
```

```JavaScript
  const response = await got(url, {
    timeout: 10000,
    retry: 2,
    headers: authHeader,
    responseType: 'json',
  });
```

```JavaScript
  const response = await got(url, {
    timeout: 10000,
    retry: 2,
    headers: authHeader,
  }).json();
```

Everything of those is returning a 415.

Questions:
* Have I overseen something?
* How can I debug the all raw headers Got is sending?
* Is Got modifying the headers unexpectedly?
* Is it a bug?

...

#### Checklist

- [x] I have read the documentation.
- [x] I have searched in Issues.
- [x] I have searched in the world wide web and also AOL groups.

**Comment by szmarczak** [comment, marked as answer]
I cannot reproduce this, but you can use Wireshark to inspect what are the differences in the requests.
